pxr/usd/usdUtils/dependencies.cpp

```diff
@@ -682,10 +702,17 @@ class _AssetLocalizer {
                     ref = ArSplitPackageRelativePathOuter(ref).first;
                 }
 
-                const std::string refAssetPath = 
+                std::string refAssetPath = 
                         SdfComputeAssetPathRelativeToLayer(
                             fileAnalyzer.GetLayer(), ref);
 
+                std::string baseName = "";
+                // UDIMs special case: remove filename and resolve dir instead
+                if (TfStringContains(refAssetPath, "<UDIM>")) {
+                    baseName = TfGetBaseName(refAssetPath);
+                    refAssetPath = TfGetPathName(refAssetPath);
+                }
+
```

Comment:
Ahh.... OK, I wasn't that familiar with this code, but I think I believe now that _this_ substitution will happen *before* the Glob matching above happens, so we don't need to replicate the logic from usdImaging.  A few requests, though?

1. For clarity, can you make a a teeny helper, `IsUDIMIdentifier()` and use it in both places you look for UDIMS?
2. Can you pull the code block below up here into a contiguous block, so the UDIM logic is less diffused, and you can move `baseName` inside the block?
3. Can you add a test?  There's a pretty good pattern already in place that you can easily copy/add.

Thanks, @ablev-dwa !

Reply:
Will do!

Reply:
Hi @spiffmon, I'm the original author of these changes. Thanks for the review so far. For (2) I'm having some difficulty finding a way to combine those changes without duplicating the sandwiched resolve and fetch checks since `baseName` can't be included for those. Do you have any suggestions, or should I just duplicate them for the sake of keeping all UDIM logic to a block?

Reply:
It seemed to me that only the Resolve() needed to be duplicated, though maybe I got that wrong. But yes, I do think that small amount of duplication is worth it to centralize the special logic. Thanks!

Reply:
I'm not very knowledgeable about how resolvers could implement FetchToLocalResolvedPath since ours does the same automatic passthrough as the default, so I just assumed some could encounter issues with a `<UDIM>` token in the name. If you still think the FetchToLocalResolvedPath is fine after the resolved dir and UDIM basename have been recombined, I'll gladly do that.

Reply:
Right... GitHub was hiding that from me.  So, FetchToLocalResolvedPath() is going to be deprecated eventually with Ar 2.0, since it is entirely counter to a big part of the point of the new API's (to be able to work with non-filesystem-based assets).  Also, the concept of "directory" is one we have explicitly refrained from defining in Ar,  because not all datastores support it.  For anything that _needs_ to be fetched locally, your TfGlob is going to fail anyways.

How about you pull the FetchToLocalResolvedPath clause into an `else` clause of the `if (IsUdimPath()` block (with an additional check that the resolvedRefFilePath is not empty ), and leave the `if (resolvedRefFilePath.empty())` check afterwards, to be encountered in both cases?  Does that make sense?

Reply:
I think your suggestion makes sense, or at least I can see a path forward now. Regardless, I'll get these changes out and we'll see if I'm on the right track! Thanks for all the info.